### PR TITLE
Gui server connection

### PR DESCRIPTION
**Pull request by AdrianHarenczyk**
This pull request makes changes to various files. Overall it adds simple Client -> Server communication. Example has elements of GUI which will not be included in final version as text field some buttons and text area. These elements helped to test connection between server and client application. Added script run_both.sh helps with running both server and client app, it was tested on two machines. It works on Ubuntu Linux for certain with gnome-terminal. You can use it to start the app.

### CODE REVIEW

File: client/src/main/java/connection/ClientIO.java [hunk 1]
```diff
@@ -32,9 +32,10 @@ void sendMessage(String message) {
         out.println(message);
     }
 
-    void getMessage() {
-        while (in.hasNextLine()) {
-            System.out.println(in.nextLine());
+    String getMessage() {
+        if (!in.hasNextLine()) {
+            out.println("Domyślna");
```
Comment (krzysieksr) [inline client/src/main/java/connection/ClientIO.java]: WTF?

File: client/src/main/java/connection/GUIServerConnection.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            handleConnectionException(e);
+        }
+    }
+
+    private void createServerConnection() throws IOException {
+        server = ClientIO.createClient(host, portNumber);
+    }
+
+    private void handleConnectionException(IOException e) {
+        e.printStackTrace();
```
Comment (krzysieksr) [inline client/src/main/java/connection/GUIServerConnection.java:44]: Please add some external logger mechanism in the future.

File: shared/src/main/java/placement/model/field/BufferField.java [hunk 3]
```diff
@@ -4,16 +4,34 @@
...
 
+    /**
+     * returns String representation of BUFFER FieldState
+     * @return
+     * certain character which indicates that this field is a buffer for
+     * some ship
+     */
     @Override
     public final String getMark() {
         return "⚓";
```
Comment (krzysieksr) [inline shared/src/main/java/placement/model/field/BufferField.java:26]: You could change anchor to another sign

**Review by krzysieksr — APPROVED** [review]

File: shared/src/main/java/placement/model/Coordinates.java [hunk 4]
```diff
@@ -17,18 +17,41 @@ private Coordinates(int coordinateX, int coordinateY) {
...
+     * @return
+     * method returns an instance of Coordinates which is set which fields as in
+     * parameters.
+     */
     public static Coordinates createCoordinates(int coordinateX, int coordinateY) {
         return new Coordinates(coordinateX, coordinateY);
     }
 
-    public final int getX() {
+   public final int getX() {
```
Comment (agaped) [inline shared/src/main/java/placement/model/Coordinates.java]: missing javadoc

**Review by agaped — CHANGES_REQUESTED** [review]
please correct

**Review by agaped — APPROVED** [review]
:+1:

**Pull request merged**